uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/components/leave-type-wizard.component.js

```diff
@@ -388,6 +484,21 @@ define([
       delete params.carry_forward_expiration_duration_switch;
     }
 
+    /**
+     * Pre-processes parameters depending on the selected leave category.
+     */
+    function preProcessParamsDependingOnLeaveCategory (params) {
+      var rules = defaultsByCategory[params.category];
+
+      if (!rules) {
```

Comment:
you can remove this guard, `_.each` will take care of it for you
```js
function preProcessParamsDependingOnLeaveCategory (params) {
  _.each(defaultsByCategory[params.category], function (default, field) {
    params[field] = default;
  });
}

Reply:
Done.